addons/web/static/src/search/search_model.js

```diff
@@ -2180,6 +2156,33 @@ export class SearchModel extends EventBus {
         this.trigger("update");
     }
 
+    /**
+     * Reconciliate the search items with the ir.filters.
+     * @private
+     */
+    _reconciliateFavorites() {
+        const irFilters = this.irFilters || [];
+        const favoriteItems = Object.values(this.searchItems).filter((o) => o.type === "favorite");
+        for (const item of favoriteItems) {
+            const filterIndex = irFilters.findIndex((i) => i.id === item.serverSideId);
```

Comment:
you can also prepare a mapping serverSideId => irFilter and use it instead of traversing irFilters multiple times